controllers/storagecluster/cephobjectstores.go

```diff
@@ -153,7 +153,13 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 					Replicated:    generateCephReplicatedSpec(initData, "metadata"),
 				},
 				Gateway: cephv1.GatewaySpec{
-					Port:      80,
+					Port:       80,
+					SecurePort: 443,
+					Service: &cephv1.RGWServiceSpec{
+						Annotations: map[string]string{
+							"service.beta.openshift.io/serving-cert-secret-name": fmt.Sprintf("%s-%s-%s", initData.Name, "cos", cephRgwTLSSecretKey),
```

Comment:
I hope this value is unique for each object-store in internal mode

Reply:
The only variable factor is the storagecluster name. 
I assume only one object store can be created per storage cluster ?  The generateCephObjectStoreName func was just returning

`Storage cluster name + "cephobjectstore"`
i shortened the `cephobjectstore` to `cos` here :) 

If my assumption is wrong or openshift is appending some unique id afterwards to the resource  name then we may need to change this something more unique

@thotz Please confirm